### PR TITLE
[Accordion]feat: 2/3

### PR DESCRIPTION
**Pull request by onmax**
## Description
Finished Accordion API and a11y

## Related Issue
#131 

## Motivation and Context
There is a major change. I removed all the `AccordionImpl` components. I don't think we need them.
The main current problem now is that as far as I know we cannot express the following logic into Vue `props`:

```ts
{
  type: AccordionType.Single;
  modelValue?: string;
} | {
  type: AccordionType.Single | AccordionType.Multiple;
  modelValue?: string[];
}
```

Basically change the type of `modelValue` depending on the type. Since we cannot do that, I just declare `modelValue`, `defaultValue` and `onValueChange` as a combination of: `string`, `undefined` and `string[]`. 

Please verify that the exports are ok.

Let me know if you see any bugs or room to improvments!

Thank you!

### CODE REVIEW

**Comment by khairulhaaziq** [comment]
Alot of things to unpack here.
1. When typing props, you need to extend the existing props if the child is another component. E.g. if the child is CollapsilbleTrigger, import the CollapsibleTriggerProps and extend them, that way you dont need to declare asChild default to false because it is already done in the bottom component(the primitives). This is what @zernonia done [here](https://github.com/radix-vue/radix-vue/pull/201) but he probably missed some in accordion.
2. We dont do onValueChange, every `value` or `open` we bind with v-model and in components we implement useVModel. You can refer to other components on how these are implemented.
3. Thanks for the refactors, we had that in planned! Actually there is a very relevant example on how to implement `single` and `multiple` props. They are `toggle group` and `select`. I think you should refer those and see how they are implemented.
4. This is just enhancer, I think we should remove the dependancies on Collapsible components and use the Primitives right away, although these are almost the same features but removing dependancies would make it much more maintainable and cleaner.

Alot to take in, overall pretty good and very nearly ready. Let me know if you need some help!

File: packages/radix-vue/src/Accordion/AccordionContent.vue
Comment (khairulhaaziq) [inline packages/radix-vue/src/Accordion/AccordionContent.vue:24]: remove the Collapsible dependancy

File: packages/radix-vue/src/Accordion/AccordionItem.vue
Comment (khairulhaaziq) [inline packages/radix-vue/src/Accordion/AccordionItem.vue:54]: we dont need computedref/setters, this can be computed only

File: packages/radix-vue/src/Accordion/AccordionItem.vue
Comment (khairulhaaziq) [inline packages/radix-vue/src/Accordion/AccordionItem.vue:102]: find another method to select the trigger. maybe put in inject as how other component does

File: packages/radix-vue/src/Accordion/AccordionHeader.vue
Comment (khairulhaaziq) [inline packages/radix-vue/src/Accordion/AccordionHeader.vue:23]: we extended AccordionHeaderProps with PrimitiveProps, so we dont need to declare asChild default to false

**Review by khairulhaaziq — CHANGES_REQUESTED** [review]
refer to the other comment

**Comment by onmax** [comment]
Great feedback. Working on it

File: packages/radix-vue/src/Accordion/AccordionRoot.vue
Comment (onmax) [reply packages/radix-vue/src/Accordion/AccordionRoot.vue:142]: But, do you want to keep the `validate` functions?

Comment (khairulhaaziq) [reply packages/radix-vue/src/Accordion/AccordionRoot.vue:142]: > But, do you want to keep the `validate` functions?

yeah validate is ok. look how small togglegroup is, try to follow that without losing any functionality/maintainability

Comment (khairulhaaziq) [reply packages/radix-vue/src/Accordion/AccordionRoot.vue:142]: > An idea: what about a composable like this:
> 
> [39a27fc](https://github.com/radix-vue/radix-vue/commit/39a27fce5ce35fa387d82aba2cfe943f58424a3f)

idk how it is useful, show how it can be used accross components

Comment (onmax) [reply packages/radix-vue/src/Accordion/AccordionRoot.vue:142]: With:

```ts
const { modelValue, changeModelValue } = useSingleOrMultipleValue(props, emits);
```

It is basically the code that we have in[ `ToggleGroupRoot`](https://github.com/onmax/radix-vue/blob/39a27fce5ce35fa387d82aba2cfe943f58424a3f/packages/radix-vue/src/ToggleGroup/ToggleGroupRoot.vue#L67)


<details><summary>And doing a quick search in VSCode, we have like 8</summary>

</details>

Comment (onmax) [reply packages/radix-vue/src/Accordion/AccordionRoot.vue:142]: So we don't repeat the same validations and the same `changeValueModel` logic

**Comment by onmax** [comment]
I think everything is ready. I will recheck everything tomorrow in case I miss something I don't see now.

I can squash the commits to clean the tree before merging

**Comment by onmax** [comment]
@khairulhaaziq you mention to remove `Collapsible`. I feel like I have to copy and paste a lot of the logic.

See [Radix implementation](https://github.com/radix-ui/primitives/blob/eca6babd188df465f64f23f3584738b85dba610e/packages/react/accordion/src/Accordion.tsx#L8):

**Comment by zernonia** [comment]
> @khairulhaaziq you mention to remove `Collapsible`. I feel like I have to copy and paste a lot of the logic.
> 
> See [Radix implementation](https://github.com/radix-ui/primitives/blob/eca6babd188df465f64f23f3584738b85dba610e/packages/react/accordion/src/Accordion.tsx#L8):

I dont recommend remove Collapsible. What's the issue that leads to the idea of removing ya?

**Comment by khairulhaaziq** [comment]
not working for me: 

https://github.com/radix-vue/radix-vue/assets/101852870/fddfcef3-d7d9-4436-acdf-1d903729e2df

**Comment by khairulhaaziq** [comment]
this has been a slow progress do to my absence, sorry. @zernonia you can take it over

File: packages/radix-vue/src/Accordion/story/AccordionSingle.story.vue
Comment (zernonia) [inline packages/radix-vue/src/Accordion/story/AccordionSingle.story.vue:111]: I think let's revert this back to `Collapsible`. As the default value for `collapsible` is `false`. this Variant here allow us to test `collapsible: true`

File: packages/radix-vue/src/Accordion/AccordionRoot.vue
Comment (zernonia) [inline packages/radix-vue/src/Accordion/AccordionRoot.vue:86]: I think you missed out passing `defaultValue` here.

causing the Accordion not open by default in uncontrollable state


```
    <AccordionRoot
        class="w-[300px] rounded-md bg-[--line-color] shadow-lg"
        type="single"
        default-value="item-3"
      >
```
image

File: packages/radix-vue/src/Accordion/AccordionRoot.vue
Comment (zernonia) [reply packages/radix-vue/src/Accordion/AccordionRoot.vue]: I think this composable is good. It handles logic for single/multiple for `modelValue`, and with validation 👍🏻.

File: packages/radix-vue/src/Accordion/AccordionItem.vue
Comment (onmax) [reply packages/radix-vue/src/Accordion/AccordionItem.vue:69]: No. If we return boolean, then the element will have `data-disabled="true"` or `data-disabled="false"`. What we need is to set `data-disabled` to the element or we remove it.

The Radix documentation states: "`data-disabled` is present when is disabled". [Docs](https://www.radix-ui.com/docs/primitives/components/accordion#item)

See [Radix code implementation](https://github.com/radix-ui/primitives/blob/main/packages/react/accordion/src/Accordion.tsx#L437)

File: packages/radix-vue/src/Accordion/story/AccordionSingle.story.vue
Comment (onmax) [reply packages/radix-vue/src/Accordion/story/AccordionSingle.story.vue:111]: True, my bad

File: packages/radix-vue/src/Accordion/AccordionRoot.vue
Comment (onmax) [reply packages/radix-vue/src/Accordion/AccordionRoot.vue:86]: I will get the value directly from the props. I think is cleaner

**Comment by onmax** [comment]
@zernonia I pushed some changes. 

[This comment ](https://github.com/radix-vue/radix-vue/pull/211#discussion_r1270205779)is the missing thing now

**Comment by khairulhaaziq** [comment]
@onmax since our reviews halt the development abit, do consider working on another component meanwhile. @zernonia i think can merge?

**Comment by onmax** [comment]
Yeah, I forgot to create a new branch for the accordion. So now I was waiting for the merge, to start working in a new component.

Nonetheless, today I will be travelling, so I can wait 😄

**Comment by zernonia** [comment]
@khairulhaaziq  @onmax  this is working well! And loves the validation messages. Clear and easily actionable!

**Review by zernonia — APPROVED** [review]

**Pull request merged**
